### PR TITLE
Add feature of upload repo or branch

### PR DESCRIPTION
**Pull request by SeanHH86**
#20 , #21

### CODE REVIEW

**Review by ganisback — DISMISSED** [review]
LGTM

**Review by ganisback — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -184,6 +184,51 @@ for item in repo_files:
...
+r = Repository(
+    repo_id="wanghh2003/ds12",
+    work_dir="/Users/hhwang/temp/ccc",
+    user_name="wanghh2003",
+    token=token,
+    repo_type="dataset"
+)
+
+# push to a new repo
+r.upload_as_new_repo(
```
Comment (Rader) [inline README.md:205]: it's better to use one single method `upload`, which create repo and branch internally if needed. 
param `auto_create_repo` can be used to control repo creation. I suggest it defaults to `true`.

Comment (SeanHH86) [reply README.md]: Merge to one method upload.

File: pycsghub/repository.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+        self.nickname = nickname
+        self.description = description
+        self.repo_type = repo_type
+        self.endpoint = endpoint
+        self.auto_create = auto_create
+        self.copy_files = copy_files
+        if self.repo_type == REPO_TYPE_DATASET:
+            self.repo_url_prefix = "datasets"
+        else:
+            self.repo_url_prefix = "models"
```
Comment (Rader) [inline pycsghub/repository.py:49]: why not support other repo types like codes and spaces

Comment (SeanHH86) [reply pycsghub/repository.py]: add repo type space and code

**Review by Rader — APPROVED** [review]

**Pull request merged**